client/src/components/Verejne/Map/index.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+import React from 'react'
+import {connect} from 'react-redux'
+import {
+  zoomSelector,
+  centerSelector,
+  entitiesSelector,
+  clustersSelector,
+  mapReferenceSelector,
+} from '../../../selectors'
+import {
+  setMapReference,
+  setMapOptions,
+  selectEntity,
+  zoomToLocation,
+} from '../../../actions/verejneActions'
+import './GoogleMap.css'
+import {map} from 'lodash'
+import ClusterMarker from './ClusterMarker'
+import {branch, compose, renderComponent, withHandlers} from 'recompose'
+import Loading from '../../Loading'
+import GoogleMap from '../../GoogleMap'
+import {withDataProviders, withRefetch} from 'data-provider'
+import {
+  entitiesProvider,
+  getEntitiesUrlFromMapReference,
+} from '../../../dataProviders/publiclyDataProviders'
+
+import type {MapOptions, Entity, MapReference, Center} from '../../../state'
+import type {MapCluster} from '../../../selectors'
+import type {Thunk, GenericAction} from '../../../types/reduxTypes'
+
+type Props = {
+  zoom: number,
+  center: [number, number],
+  entities: Array<Entity>,
+  setMapOptions: (mapOptions: MapOptions) => GenericAction<MapOptions, MapOptions>,
+  setMapReference: (mapReference: MapReference) => GenericAction<MapReference, MapReference>,
+  selectEntity: (entity: Entity) => Thunk,
+  zoomToLocation: (center: Center) => Thunk,
+  refetch: () => void,
+  clusters: Array<MapCluster>,
+  onGoogleApiLoaded: ({map: MapReference}) => any,
+  onChange: (options: MapOptions) => any,
+}
+
+const Map = ({
+  zoom,
+  center,
+  entities,
+  setMapOptions,
+  setMapReference,
+  selectEntity,
+  zoomToLocation,
+  refetch,
+  clusters,
+  onGoogleApiLoaded,
+  onChange,
+}: Props) => {
+  return (
+    <div className="google-map-wrapper">
+      <GoogleMap
+        center={center}
+        zoom={zoom}
+        onGoogleApiLoaded={onGoogleApiLoaded}
+        onChange={onChange}
+      >
+        {map(clusters, (e, i) => (
+          <ClusterMarker
+            key={i}
+            entity={e}
+            selectEntity={selectEntity}
+            zoom={zoom}
+            zoomToLocation={zoomToLocation}
+            lat={e.lat}
+            lng={e.lng}
+          />
+        ))}
+      </GoogleMap>
+    </div>
+  )
+}
+
+export default compose(
+  connect(
```

Comment:
please split this connects as written in previous comment (now outdated as the original file was split)